pkg/entities/record.go

```diff
@@ -101,91 +102,41 @@ func (d *dataRecord) PrepareRecord() (uint16, error) {
 
 func (d *dataRecord) AddInfoElement(element *InfoElement, val interface{}) (uint16, error) {
 	d.fieldCount++
-	var bytesToAppend []byte
-	if element.Len != VariableLength {
-		bytesToAppend = make([]byte, element.Len)
-	} else {
-		bytesToAppend = make([]byte, 0)
-	}
+	length := d.buff.Len()
+	var err error
 	switch dataType := element.DataType; dataType {
-	case Unsigned8:
-		v, ok := val.(uint8)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type uint8")
-		}
-		bytesToAppend[0] = v
-	case Unsigned16:
-		v, ok := val.(uint16)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type uint16")
-		}
-		binary.BigEndian.PutUint16(bytesToAppend, v)
-	case Unsigned32:
-		v, ok := val.(uint32)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type uint32")
-		}
-		binary.BigEndian.PutUint32(bytesToAppend, v)
-	case Unsigned64:
-		v, ok := val.(uint64)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type uint64")
-		}
-		binary.BigEndian.PutUint64(bytesToAppend, v)
-	case Signed8:
-		v, ok := val.(int8)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type int8")
-		}
-		bytesToAppend[0] = byte(v)
-	case Signed16:
-		v, ok := val.(int16)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type int16")
-		}
-		binary.BigEndian.PutUint16(bytesToAppend, uint16(v))
-	case Signed32:
-		v, ok := val.(int32)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type int32")
-		}
-		binary.BigEndian.PutUint32(bytesToAppend, uint32(v))
-	case Signed64:
-		v, ok := val.(int64)
-		if !ok {
-			return 0, fmt.Errorf("val argument is not of type int64")
-		}
-		binary.BigEndian.PutUint64(bytesToAppend, uint64(v))
+	case Unsigned8,  Unsigned16, Unsigned32, Unsigned64, Signed8, Signed16, Signed32, Signed64:
+		err = util.Encode(&d.buff, val)
```

Comment:
how are we checking if the val input argument is same type as we expected? util.Encode seems to take in any argument.

Reply:
I will need to handle each case separately then. Thanks for pointing it out.